### DISCUSSION TITLE
cv2 error in harcascade file

### DISCUSSION THREAD

**Discussion by Naman065Malik** [datasciencecompleteguide/Machine-Learning#6] [Q&A]
I am trying to make a face lock system for my AI (jarvis). Right now I am collecting samples of my face but there is a error:(-215:Assertion failed) !empty() in function 'cv::CascadeClassifier::detectMultiScale' . I tried to give exact location of harcascade_frontalface_default But pycharm is still giving error.

![Screenshot 2021-06-22 235902](https://user-images.githubusercontent.com/86320160/122980164-fb160e00-d3b5-11eb-906a-48f1873b1115.png)

Traceback (most recent call last):
  File "C:\Users\DELL\PycharmProjects\jarvisfulltry\samplegenerator.py", line 14, in <module>
    faces = detector.detectMultiScale(converted_image, 1.3, 5)
cv2.error: OpenCV(4.5.2) C:\Users\runneradmin\AppData\Local\Temp\pip-req-build-ttbyx0jz\opencv\modules\objdetect\src\cascadedetect.cpp:1689: error: (-215:Assertion failed) !empty() in function 'cv::CascadeClassifier::detectMultiScale'
![Screenshot 2021-06-23 000423](https://user-images.githubusercontent.com/86320160/122980718-9effb980-d3b6-11eb-8a68-d4e20362ce29.png)
Please help me

**Comment by datasciencecompleteguide** [comment]
> 
> 
> I am trying to make a face lock system for my AI (jarvis). Right now I am collecting samples of my face but there is a error:(-215:Assertion failed) !empty() in function 'cv::CascadeClassifier::detectMultiScale' . I tried to give exact location of harcascade_frontalface_default But pycharm is still giving error.
> 
> ![Screenshot 2021-06-22 235902](https://user-images.githubusercontent.com/86320160/122980164-fb160e00-d3b5-11eb-906a-48f1873b1115.png)
> 
> Traceback (most recent call last):
> File "C:\Users\DELL\PycharmProjects\jarvisfulltry\samplegenerator.py", line 14, in
> faces = detector.detectMultiScale(converted_image, 1.3, 5)
> cv2.error: OpenCV(4.5.2) C:\Users\runneradmin\AppData\Local\Temp\pip-req-build-ttbyx0jz\opencv\modules\objdetect\src\cascadedetect.cpp:1689: error: (-215:Assertion failed) !empty() in function 'cv::CascadeClassifier::detectMultiScale'
> ![Screenshot 2021-06-23 000423](https://user-images.githubusercontent.com/86320160/122980718-9effb980-d3b6-11eb-8a68-d4e20362ce29.png)
> Please help me


In the 7th line of your code you have given the file name as
"haarcascade_frontal_default.xml"
but the name should be 
"**haarcascade_frontalface_default.xml**" as per the code you have shared to me.
